test/communication_test.go

```diff
@@ -43,22 +43,21 @@ func TestMain(m *testing.M) {
 
 func runTestGrpcClient(schemaType pb.SchemaType, schemaVariant pb.SchemaVariant,
 	protocolParams client.ProtocolParams) error {
-	c, err := client.NewProtocolClient(testGrpcServerEndpont, schemaType, schemaVariant)
-	if err != nil {
-		return err
-	}
-	c.ExecuteProtocol(protocolParams)
-	return nil
+	c := client.NewClient(testGrpcServerEndpont)
```

Comment:
Same here - instantiate target client directly and invoke its method that you want to test.